### DISCUSSION TITLE
draw.io / diagrams.net in MS Teams

### DISCUSSION THREAD

**Discussion by lrarbin** [jgraph/drawio#2428] [Ideas]
Hello,
We've just started using the draw.io app in MS Teams:
https://appsource.microsoft.com/en-gb/product/office/WA200003444?tab=DetailsAndSupport

We've already come across a couple of issues.
- We've found that in Dark Mode in MS Teams the draw.io interface can be difficult (nearly impossible) to read. (see attached)
- Selecting a draw.io file or where to save it is confusing. It Team Channel Chats they should default to the Channel Folder in Teams for the Team. In Group / Peer to Peer Chats the files should default to the users Shared Folder on OneDrive.

Thanks
Laurence

![2021-11-19 10_42_51-Window](https://user-images.githubusercontent.com/8376969/142610409-45da5482-4a98-43eb-84ef-cf0d752c7181.png)

**Comment by m-mohamedin** [comment]
Thanks for your feedback.
We'll work on fixing the dark mode. Regarding default storage location, when Sharepoint button is clicked in Teams channel, it opens that Team folder on Sharepoint directly. But, for P2P chat, we cannot decide which folder to choose since it could be more than one folder with shared contents. Please correct me or give us an example of the default Shared Folder on OneDrive

Thanks

**Reply by lrarbin** [reply]
Hello Mohamed,
Yes the SharePoint button is opening to a list of all SharePoint Sites (or at least the top 100 or similar), but not the one related to Team Channel you are currently in. I've attached a screenshot.
![2021-11-19 13_41_34-Clipboard](https://user-images.githubusercontent.com/8376969/142632245-75c99b62-bc53-413b-87b1-75121bf0b26f.png)

And regards Group / Peer to Peer Chat. Shared files should be saved in the 'Microsoft Teams Chat Files' BUT I think the 'draw.io add-in' would need to also set the sharing permissions for the file to the members of the chat.
Here's an example of the Microsoft Teams Chat Files.
![2021-11-19 13_17_44-Window](https://user-images.githubusercontent.com/8376969/142632492-8a09ffbc-cb0e-4f2b-ae9b-dc8a0d1d31a1.png)

NB. We are using the Enterprise Version of MS Teams / OneDrive (Microsoft365), not the consumer version.

Thanks for the help.
Laurence

**Reply by m-mohamedin** [reply]
Sharepoint button on the Teams opens the teams folder, but in the Chat it just open all Sharepoint sites since don't get information about the team site. Which chat are you referring to (P2P or something else?)
![Screen Shot 2021-11-21 at 2 43 26 PM](https://user-images.githubusercontent.com/24944765/142776699-e0fda813-8d15-4e49-8d65-f13ad231ede2.png)

**Reply by lrarbin** [reply]
Hello,
Sorry for the massive delay in replying. This is in Channel Chat (within a Team). I've done some more investigating. In my 'test' tenant, the draw.io works as I would expect. But in our production Tenant, I'm getting the behaviour when ALL the MS Teams are listed, I'm not directed to the SharePoint Document Library for the Team.

In AAD Enterprise Apps the permissions are different in my Test Tenant and the Production Tenant:
Test Tenant
![image](https://user-images.githubusercontent.com/8376969/149781160-13d5423a-d7ae-424d-a25e-96240d2ee4a3.png)

Production Tenant
![image](https://user-images.githubusercontent.com/8376969/149781326-d03507d5-aa69-4d2c-99c6-3dc6c09dac80.png)

Our Production Tenant is on a Multi Geo Tenant, so SharePoints Sites for different locations 'can' appear in different SharePoint locations, however 99% of our Teams are configured for EMEA, so I don't think this can be the issue.

Thanks
Laurence

**Reply by m-mohamedin** [reply]
Can you try deleting the application such that it asks for the consent again? Does it help?
